lib/common/ipc_client.c

```diff
@@ -1232,23 +1237,33 @@ internal_ipc_get_reply(crm_ipc_t *client, int request_id, int ms_timeout,
         }
 
         hdr = (pcmk__ipc_header_t *)(void*) client->buffer;
-        reply = pcmk__xml_parse(crm_ipc_buffer(client));
+        xml = pcmk__xml_parse(crm_ipc_buffer(client));
 
         if (hdr->qb.id < request_id) {
             crm_err("Discarding old reply %d (need %d)", hdr->qb.id, request_id);
-            crm_log_xml_notice(reply, "OldIpcReply");
+            crm_log_xml_notice(xml, "OldIpcReply");
         } else if (hdr->qb.id > request_id) {
             crm_err("Discarding newer reply %d (need %d)", hdr->qb.id, request_id);
-            crm_log_xml_notice(reply, "ImpossibleReply");
+            crm_log_xml_notice(xml, "ImpossibleReply");
             pcmk__assert(hdr->qb.id <= request_id);
         } else {
             /* Got the reply we were expecting. */
             break;
         }
-    } while (time(NULL) < timeout);
+    } while (time(NULL) < timeout || (timeout == 0 && *bytes == -EAGAIN));
+
+    if (*bytes > 0) {
+        crm_trace("Received %zd-byte reply %d to %s IPC %d: %.100s",
+                  *bytes, hdr->qb.id, client->server_name, request_id,
+                  crm_ipc_buffer(client));
 
-    if (*bytes < 0) {
+        if (reply != NULL) {
+            *reply = xml;
+        }
+    } else if (*bytes < 0) {
         rc = (int) -*bytes; // System errno
+        crm_trace("No reply to %s IPC %d: rc=%d", client->server_name,
```

Comment:
This matters less since it's tracing, but it's nice to have a `pcmk_rc_str()` of the `rc`.